filters.im

```diff
@@ -1524,6 +1524,102 @@ i_diff_image(i_img *im1, i_img *im2, double mindist) {
   return out;
 }
 
+/*
+=item i_rgbdiff_image(im1, im2)
+
+Creates a new image that is black, except where the pixel in im2 is
+different from im1, where it is the arithmetical difference to im2 per
+color.
+
+=cut
+*/
+
+i_img *
+i_rgbdiff_image(i_img *im1, i_img *im2) {
+  i_img *out;
+  int outchans, diffchans;
+  i_img_dim xsize, ysize;
+  dIMCTXim(im1);
+
+  i_clear_error();
+  if (im1->channels != im2->channels) {
+    i_push_error(0, "different number of channels");
+    return NULL;
+  }
+
+  outchans = diffchans = im1->channels;
+  if (outchans == 1 || outchans == 3)
+    ++outchans;
+
+  xsize = i_min(im1->xsize, im2->xsize);
+  ysize = i_min(im1->ysize, im2->ysize);
+
+  out = i_sametype_chans(im1, xsize, ysize, outchans);
+
+  if (im1->bits == i_8_bits && im2->bits == i_8_bits) {
```

Comment:
Ideally this should use the preprocessor to avoid having to duplicate the code between 8-bit and wider sample images.

See Imager::Preprocess for some limited documentation and scale.im is probably the best example of existing code using it.

Reply:
Great, that makes the code shorter indeed. Addressed in my commit 9bcbb2ad8dab0d5475cbf93e195ca2a7d7b891fc.